src/OrleansRuntime/Hosting/DefaultSiloServices.cs

```diff
@@ -61,8 +63,15 @@ internal static void AddDefaultServices(IServiceCollection services)
             services.TryAddTransient(typeof(IStreamSubscriptionObserver<>), typeof(StreamSubscriptionObserverProxy<>));
 
             services.TryAddSingleton<StatisticsProviderManager>();
+
             services.TryAddSingleton<StorageProviderManager>();
+            services.TryAddSingleton<IKeyedServiceCollection<string, IStorageProvider>>(sp => sp.GetRequiredService<StorageProviderManager>());
```

Comment:
Doesn't need to change in this PR, but there's an extension method called `TryAddFromExisting` which essentially does this.